msteams-platform/toolkit/deploy-Teams-app-to-container-service.md

```diff
@@ -0,0 +1,304 @@
+    
+    helm repo add jetstack https://charts.jetstack.io
+    helm repo update
+    helm install cert-manager jetstack/cert-manager --namespace $NAMESPACE --set installCRDs=true --set nodeSelector."kubernetes\.io/os"=linux
+    ```
+
+1. Update the DNS for the ingress public IP and get the ingress endpoint.
+
+    ```bash
+    > kubectl get services --namespace $NAMESPACE -w ingress-nginx-controller
+    
+    NAME TYPE CLUSTER-IP EXTERNAL-IP PORT(S)
+    ingress-nginx-controller LoadBalancer $CLUSTER_IP $EXTERNAL_IP 80:32514/TCP,443:32226/TCP
+    
+    > PUBLICIPID=$(az network public-ip list --query "[?ipAddress!=null]|[?contains(ipAddress, '$EXTERNAL_IP')].[id]" --output tsv)
+    > az network public-ip update --ids $PUBLICIPID --dns-name $DNSLABEL
+    > az network public-ip show --ids $PUBLICIPID --query "[dnsSettings.fqdn]" --output tsv
+    
+    $DNSLABEL.$REGION.cloudapp.azure.com
+    ```
+
+### Provision resources with Teams Toolkit
+
+You can use the provision command in Teams Toolkit to create a Teams app with bot capability, incorporating the Azure Bot Service and the Entra ID for authentication. Update the sample code to ensure compatibility with your Azure Kubernetes Service.
+
+1. Update the `BOT_DOMAIN` value in `env/.env.${envName}` with your FQDN.
+
+1. Update the `arm/deploy` action within `teamsapp.yml` to ensure that Teams Toolkit provision an Azure Bot Service during the execution of the `provision` command.
+
+    ```bash
+    - uses: arm/deploy 
+      with:
+        subscriptionId: ${{AZURE_SUBSCRIPTION_ID}} 
+        resourceGroupName: ${{AZURE_RESOURCE_GROUP_NAME}} 
+        templates:
+          - path: ./infra/botRegistration/azurebot.bicep
+            parameters: ./infra/botRegistration/azurebot.parameters.json
+            deploymentName: Create-resources-for-bot
+        bicepCliVersion: v0.9.1
+    ```
+
+1. Run the `provision` command in Teams Toolkit.
+
+1. After provisioning, locate the `BOT_ID` in `env/.env.${envName}` file and the encrypted `SECRET_BOT_PASSWORD` in `env/.env.${envName}.user` file. To obtain the actual value of `BOT_PASSWORD`, select the Decrypt secret annotation.
+
+1. To create a Kubernetes secret that contains `BOT_ID` and `BOT_PASSWORD`, save the key value pair in the `./deploy/.env.dev-secrets`file and execute the command to provision the secret.
+
+    ```bash
+    kubectl create secret generic dev-secrets --from-env-file ./deploy/.env.dev-secrets -n $NAMESPACE
+    ```
+
+### Apply the deployment
+
+The sample includes a deployment file, `deploy/sso-bot.yaml`, for your reference. Update the placeholders in this file before you apply it:
+
+1. Update the `<image>` placeholder with your image. For example, `myacr.azurecr.io/sso-bot:latest`.
+
+1. Update the `<hostname>` with your ingress FQDN.
+
+1. Update the `<email>` with your email address for generating TLS certificate.
```

Comment:
We will remove the email placeholder from the yaml file. Please remove this step, thanks.

Reply:
Updated.